packages/nodejs/src/typeTranslation.ts

```diff
@@ -235,6 +244,116 @@ function transBaker(
     };
 }
 
+function translateChainParametersCommon(
+    params: v2.ChainParametersV1 | v2.ChainParametersV0
+): v1.ChainParametersCommon {
+    return {
+        electionDifficulty: transAmountFraction(
+            params.electionDifficulty?.value
+        ),
+        euroPerEnergy: unwrap(params.euroPerEnergy?.value),
+        microGTUPerEuro: unwrap(params.microCcdPerEuro?.value),
+        accountCreationLimit: unwrap(params.accountCreationLimit?.value),
+        foundationAccount: v1.AccountAddress.fromBytes(
+            Buffer.from(unwrap(params.foundationAccount?.value))
+        ).address,
+    };
+}
+
+function translateCommissionRange(
+    range: v2.InclusiveRangeAmountFraction | undefined
+): v1.InclusiveRange<number> {
+    return {
+        min: transAmountFraction(range?.min),
+        max: transAmountFraction(range?.max),
+    };
+}
+
+function translateRewardParametersCommon(
+    params: v2.ChainParametersV1 | v2.ChainParametersV0
+): v1.RewardParametersCommon {
+    return {
+        transactionFeeDistribution: {
+            baker: transAmountFraction(
+                params.transactionFeeDistribution?.baker
+            ),
+            gasAccount: transAmountFraction(
+                params.transactionFeeDistribution?.gasAccount
+            ),
+        },
+        gASRewards: {
+            baker: transAmountFraction(params.gasRewards?.baker),
+            finalizationProof: transAmountFraction(
+                params.gasRewards?.finalizationProof
+            ),
+            accountCreation: transAmountFraction(
+                params.gasRewards?.accountCreation
+            ),
+            chainUpdate: transAmountFraction(params.gasRewards?.chainUpdate),
+        },
+    };
+}
+
+function translateMintRate(mintRate: v2.MintRate | undefined): number {
+    return unwrap(mintRate?.mantissa) * 10 ** (-1 * unwrap(mintRate?.exponent));
+}
+
+function transPoolPendingChange(
+    change: v2.PoolPendingChange | undefined
+): v1.BakerPoolPendingChange {
+    switch (change?.change?.oneofKind) {
+        case 'reduce': {
+            return {
+                pendingChangeType:
+                    BakerPoolPendingChangeType.ReduceBakerCapital,
+                effectiveTime: new Date(
+                    Number(unwrap(change.change.reduce.effectiveTime?.value))
+                ),
```

Comment:
`transTimestamp()`? Note that `transTimestamp()` multiplies with 1000 but should we also do this here?

Reply:
I've changed it to use transTimestamp, but it should not multiply with 1000. (But checking the types/comments, it should never do that in the used cases, the timestamp should be in milliseconds and the date constructor expects a number argument to be in milliseconds)

Reply:
(Okay I've broken a test now. I'll look at it tomorrow)